### PR TITLE
Add parsing method to bulk response

### PR DESCRIPTION
**Pull request by tlrx**
This pull request adds the `fromXContent()` parsing method to BulkResponse.

### CODE REVIEW

**Comment by tlrx** [comment]
This is ready to review.

File: core/src/main/java/org/elasticsearch/action/bulk/BulkResponse.java [hunk 1]
```diff
@@ -141,4 +156,61 @@ public void writeTo(StreamOutput out) throws IOException {
...
+        builder.endArray();
+        builder.endObject();
+        return builder;
+    }
+
+    public static BulkResponse fromXContent(XContentParser parser) throws IOException {
+        XContentParser.Token token = parser.nextToken();
+        ensureExpectedToken(XContentParser.Token.START_OBJECT, token, parser::getTokenLocation);
+
+        long took = 0;
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/action/bulk/BulkResponse.java]: nit: shall we initialize it to -1?

File: core/src/main/java/org/elasticsearch/action/bulk/BulkResponse.java [hunk 2]
```diff
@@ -141,4 +156,61 @@ public void writeTo(StreamOutput out) throws IOException {
...
+        while ((token = parser.nextToken()) != XContentParser.Token.END_OBJECT) {
+            if (token == XContentParser.Token.FIELD_NAME) {
+                currentFieldName = parser.currentName();
+            } else if (token.isValue()) {
+                if (TOOK.equals(currentFieldName)) {
+                    took = parser.longValue();
+                } else if (INGEST_TOOK.equals(currentFieldName)) {
+                    ingestTook = parser.longValue();
+                } else if (ERRORS.equals(currentFieldName) == false) {
+                    throwUnknownField(currentFieldName, parser.getTokenLocation());
```
Comment (javanna) [inline core/src/main/java/org/elasticsearch/action/bulk/BulkResponse.java]: we don't parse errors as we don't have a field for it in the response?

**Review by javanna — APPROVED** [review]
LGTM

File: core/src/main/java/org/elasticsearch/action/bulk/BulkResponse.java (hunk 1 above)
Comment (tlrx) [inline core/src/main/java/org/elasticsearch/action/bulk/BulkResponse.java]: Yes, that would be aligned with ingest took millis.

File: core/src/main/java/org/elasticsearch/action/bulk/BulkResponse.java (hunk 2 above)
Comment (tlrx) [inline core/src/main/java/org/elasticsearch/action/bulk/BulkResponse.java]: Yes, `errors` is the output of the BulkResponse.hasFailures() method which simply iterates over bulk item and checks for failed item.

**Comment by tlrx** [comment]
@elasticmachine retest this please

**Comment by tlrx** [comment]
Thanks @javanna !

**Pull request merged**